# HG changeset patch
# User Serviscope Minor <serviscope_minor@verybigfrog.com>
# Date 1407248816 -3600
#      Tue Aug 05 15:26:56 2014 +0100
# Node ID 538f6492f21cd5646bcb087d0b077dcea30d87c4
# Parent  8ac4ab4ae5f40ff8074cd0df43b0bae260575466
Honour the title.color property when plotting with gnuplot (bug #42927).

* scripts/plot/util/private/__go_draw_axes__.m: extract and use the title
  text colorspec when creating the title command.

diff --git a/scripts/plot/util/private/__go_draw_axes__.m b/scripts/plot/util/private/__go_draw_axes__.m
--- a/scripts/plot/util/private/__go_draw_axes__.m
+++ b/scripts/plot/util/private/__go_draw_axes__.m
@@ -148,20 +148,21 @@ function __go_draw_axes__ (h, plot_strea
   fputs (plot_stream, "unset x2tics;\n");
   fputs (plot_stream, "unset x2tics;\n");
 
   if (! isempty (axis_obj.title))
     t = get (axis_obj.title);
     if (isempty (t.string))
       fputs (plot_stream, "unset title;\n");
     else
+      colorspec = get_text_colorspec (get (axis_obj.title, "color"), mono);
       [tt, f, s] = __maybe_munge_text__ (enhanced, t, "string");
       fontspec = create_fontspec (f, s, gnuplot_term);
-      fprintf (plot_stream, "set title \"%s\" %s %s;\n",
-               undo_string_escapes (tt), fontspec,
+      fprintf (plot_stream, "set title \"%s\" %s %s %s;\n",
+               undo_string_escapes (tt), fontspec, colorspec, 
                __do_enhanced_option__ (enhanced, t));
     endif
   endif
 
   if (! isempty (axis_obj.xlabel))
     t = get (axis_obj.xlabel);
     angle = t.rotation;
     colorspec = get_text_colorspec (axis_obj.xcolor, mono);
